File: after-midnight-backend/src/main/java/com/aftermidnight/service/AuthService.java
package com.aftermidnight.service;

import com.aftermidnight.dto.AuthResponse;
import com.aftermidnight.dto.LoginRequest;
import com.aftermidnight.dto.RegisterRequest;

public interface AuthService {
    AuthResponse register(RegisterRequest request);
    AuthResponse login(LoginRequest request);
}

File: frontend/src/services/auth.service.ts
import { api } from '@/lib/api';
import { LoginRequest, RegisterRequest, AuthResponse } from '@/types/auth';
import { ApiResponse } from '@/types/common';

export const AuthService = {
    async register(data: RegisterRequest): Promise<AuthResponse> {
        const response = await api.post<any, ApiResponse<AuthResponse>>('/auth/register', data);
        return response.data;
    },

    async login(data: LoginRequest): Promise<AuthResponse> {
        const response = await api.post<any, ApiResponse<AuthResponse>>('/auth/login', data);
        return response.data;
    }
};
